### PR TITLE
Autocomplete

### PR DESCRIPTION
**Pull request by cryogenian**
- Fixed couple bugs with incorrect foreign functions names
- Added autocomplete 
- pursuit 
- **ace** API has been changed, updated bindings of `onChange`
- Switched from `^` to `~` for **ace-builds** to prevent ↥ . 
- reformatted files I worked with to make lines a bit shorter 

@jonsterling Please review

### CODE REVIEW

**Comment by jonsterling** [comment]
@cryogenian it looks like there is some kind of build problem, possibly related to the removal of the `docs` task here?: https://github.com/cryogenian/purescript-ace/commit/5df76011f9489043e351c34db9b51bc65b277c71#diff-b9e12334e9eafd8341a6107dd98510c9L31

**Comment by cryogenian** [comment]
Sorry about this. Fixed

**Comment by jonsterling** [comment]
Thanks, no problem! I take it the reason for removing the docs folder is that the documentation will be deployed to Pursuit anyway?

File: src/Ace/Types.purs [hunk 1]
```diff
@@ -16,31 +14,37 @@ type AnchorEvent =
...
+
+instance documentEventTypeIsForeign :: IsForeign DocumentEventType where
+  read = readString >=> readDocumentEventType
+
+newtype DocumentEvent = DocumentEvent
+  { action :: DocumentEventType
+  , start :: Position
+  , end :: Position
+  , lines :: Array String
+  }
```
Comment (jonsterling) [inline src/Ace/Types.purs:38]: So these fields are all guaranteed to be in the object?

Comment (cryogenian) [inline src/Ace/Types.purs:38]: Yes, they are.

**Comment by cryogenian** [comment]
Yes, I've removed `docs` task because docs will be deployed on pursuit

**Comment by jonsterling** [comment]
:+1: looks good to me!

**Comment by cryogenian** [comment]
Thanks for review!

**Pull request merged**